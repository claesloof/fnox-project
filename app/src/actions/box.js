import axios from 'axios';

import {
  FETCH_BOXES_REQUEST,
  FETCH_BOXES_SUCCESS,
  FETCH_BOXES_FAILURE,
  REGISTER_BOX_REQUEST,
  REGISTER_BOX_SUCCESS,
  REGISTER_BOX_FAILURE
} from '../constants/actionTypes';

const API_ROOT = 'http://localhost:8080';

export function register(name, weight, color, destination, multiplier) {
  const cost = weight * multiplier;
  const data = {
    name: name,
    weight: weight,
    color: color,
    destination: destination,
    cost: cost
  }

  return dispatch => {
    dispatch({type: REGISTER_BOX_REQUEST});
    return axios({
      method: 'POST',
      url: API_ROOT + '/boxes',
      data: data,
      headers: {
        'Content-Type': 'application/json'
      }
    }).then(response => {
      dispatch({type: REGISTER_BOX_SUCCESS, payload: response.data});
    }).catch(error => {
      dispatch({type: REGISTER_BOX_FAILURE, payload: error});
    });
  }
}

export function fetch() {
  return dispatch => {
    dispatch({type: FETCH_BOXES_REQUEST});
    return axios({
      method: 'GET',
      url: API_ROOT + '/boxes'
    }).then(response => {
      dispatch({type: FETCH_BOXES_SUCCESS, payload: response.data});
    }).catch(error => {
      dispatch({type: FETCH_BOXES_FAILURE, payload: error});
    });
  }
}
