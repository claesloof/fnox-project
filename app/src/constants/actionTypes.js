export const FETCH_BOXES_REQUEST = "FETCH_BOXES_REQUEST";
export const FETCH_BOXES_SUCCESS = "FETCH_BOXES_SUCCESS";
export const FETCH_BOXES_FAILURE = "FETCH_BOXES_FAILURE"

export const INITIALIZE_FORM = "INITIALIZE_FORM";

export const REGISTER_BOX_REQUEST = "REGISTER_BOX_REQUEST";
export const REGISTER_BOX_SUCCESS = "REGISTER_BOX_SUCCESS";
export const REGISTER_BOX_FAILURE = "REGISTER_BOX_FAILURE";

export const UPDATE_BOX_FIELD = "UPDATE_BOX_FIELD";
