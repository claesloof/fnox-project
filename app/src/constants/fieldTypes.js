
export const DESTINATIONS = [
  {
    id: "sweden",
    label: "Sweden",
    multiplier: 1.3
  },
  {
    id: "china",
    label: "China",
    multiplier: 4
  },
  {
    id: "brazil",
    label: "Brazil",
    multiplier: 8.6
  },
  {
    id: "australia",
    label: "Australia",
    multiplier: 7.2
  }
];


export const FIELD_DATA = {
  nameField: {
    id: "nameField",
    defaultValue: "",

  },
  weightField: {
    id: "weightField",
    defaultValue: 0
  },

  colorField: {
    id: "colorField",
    defaultValue: "#ffffff"
  },
  destinationField: {
    id: "destinationField",
    defaultValue: DESTINATIONS[0].id
  }
};
