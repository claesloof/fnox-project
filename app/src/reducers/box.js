import {
  FETCH_BOXES_REQUEST,
  FETCH_BOXES_SUCCESS,
  FETCH_BOXES_FAILURE,
  REGISTER_BOX_REQUEST,
  REGISTER_BOX_SUCCESS,
  REGISTER_BOX_FAILURE,
} from '../constants/actionTypes';

import {
  DESTINATIONS
} from '../constants/fieldTypes';

const box = (state = {}, action) => {
  switch(action.type) {
    case FETCH_BOXES_REQUEST:
      return {
        ...state,
        fetchInProgress: true
      }
    case FETCH_BOXES_SUCCESS:
      return {
        ...state,
        // fetchInProgress: false,
        boxes: action.payload
      }
    case FETCH_BOXES_FAILURE:
      return {
        ...state,
        fetchInProgress: false,
        error: action.payload
      }
    case REGISTER_BOX_REQUEST:
      return {
        ...state,
        postInProgress: true
      };
    case REGISTER_BOX_SUCCESS:
      return {
        ...state,
        postInProgress: false
      };
    case REGISTER_BOX_FAILURE:
      return {
        ...state,
        error: action.payload,
        postInProgress: false
      };
    default:
      return state;
  }
}

export default box;
