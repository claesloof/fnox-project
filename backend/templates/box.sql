# ************************************************************
# Sequel Pro SQL dump
# Version 5224
#
# http://www.sequelpro.com/
# https://github.com/sequelpro/sequelpro
#
# Host: 127.0.0.1 (MySQL 8.0.12)
# Database: boxes_db
# Generation Time: 2018-09-21 13:42:12 +0000
# ************************************************************


/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;
/*!40101 SET @OLD_CHARACTER_SET_RESULTS=@@CHARACTER_SET_RESULTS */;
/*!40101 SET @OLD_COLLATION_CONNECTION=@@COLLATION_CONNECTION */;
/*!40101 SET NAMES utf8 */;
SET NAMES utf8mb4;
/*!40014 SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0 */;
/*!40101 SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='NO_AUTO_VALUE_ON_ZERO' */;
/*!40111 SET @OLD_SQL_NOTES=@@SQL_NOTES, SQL_NOTES=0 */;


# Dump of table box
# ------------------------------------------------------------

DROP TABLE IF EXISTS `box`;

CREATE TABLE `box` (
  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
  `name` varchar(40) CHARACTER SET utf8mb4 COLLATE utf8mb4_0900_ai_ci DEFAULT '',
  `weight` double DEFAULT NULL,
  `color` varchar(10) CHARACTER SET utf8mb4 COLLATE utf8mb4_0900_ai_ci DEFAULT '',
  `destination` varchar(40) CHARACTER SET utf8mb4 COLLATE utf8mb4_0900_ai_ci DEFAULT '',
  `cost` double DEFAULT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;




/*!40111 SET SQL_NOTES=@OLD_SQL_NOTES */;
/*!40101 SET SQL_MODE=@OLD_SQL_MODE */;
/*!40014 SET FOREIGN_KEY_CHECKS=@OLD_FOREIGN_KEY_CHECKS */;
/*!40101 SET CHARACTER_SET_CLIENT=@OLD_CHARACTER_SET_CLIENT */;
/*!40101 SET CHARACTER_SET_RESULTS=@OLD_CHARACTER_SET_RESULTS */;
/*!40101 SET COLLATION_CONNECTION=@OLD_COLLATION_CONNECTION */;
