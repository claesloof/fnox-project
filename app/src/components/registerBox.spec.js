import React from 'react';
import configureMockStore from 'redux-mock-store';
import fetchMock from 'fetch-mock'
import thunk from 'redux-thunk'
import { mount, shallow, render } from 'enzyme';

import { RegisterBox } from './RegisterBox';
import * as actions from './../actions/box';
import * as types from './../constants/actionTypes';

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares);

const mockBox = {
  name: "Slim Shady",
  weight: 2,
  color: "ffff00",
  destination: "sweden",
  multiplier: 1.2,
  cost: 2.4
}

describe('RegisterBox', () => {
  afterEach(() => {
    fetchMock.reset()
    fetchMock.restore()
  });

  it('should render correctly', () => {
    const wrapper = shallow(<RegisterBox />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should set nameField.validate to true if correct value', () => {
    const wrapper = shallow(<RegisterBox/>);
    const value = "MGK stood no chance";
    const nameField = wrapper.find(".add-box-form__text").at(0);

    nameField.simulate("change", {target: {name: "nameField", value: value}});
    expect(wrapper.state('nameField').validated).toEqual(true);
  })

  it('should set nameField.validate to false if incorrect value', () => {
    const wrapper = shallow(<RegisterBox/>);
    const value = "";
    const nameField = wrapper.find(".add-box-form__text").at(0);

    nameField.simulate("change", {target: {name: "nameField", value: value}});
    expect(wrapper.state('nameField').validated).toEqual(false);
  })

  it('should set weightField.validate to true if correct value', () => {
    const wrapper = shallow(<RegisterBox/>);
    const value = "2";
    const weightField = wrapper.find(".add-box-form__text").at(1);

    weightField.simulate("change", {target: {name: "weightField", value: value}});
    expect(wrapper.state('weightField').validated).toEqual(true);
  })

  it('should set weightField.validate to false if incorrect value', () => {
    const wrapper = shallow(<RegisterBox/>);
    const value = "wops";
    const weightField = wrapper.find(".add-box-form__text").at(1);

    weightField.simulate("change", {target: {name: "weightField", value: value}});
    expect(wrapper.state('weightField').validated).toEqual(false);
  })

  it('should set colorField.validate to true if correct value', () => {
    const wrapper = shallow(<RegisterBox/>);
    const value = "#ffff00";
    const colorField = wrapper.find(".add-box-form__color").at(0);

    colorField.simulate("change", {target: {name: "colorField", value: value}});
    expect(wrapper.state('colorField').validated).toEqual(true);
  })

  it('should set colorField.validate to false if incorrect value', () => {
    const wrapper = shallow(<RegisterBox/>);
    const value = "#ffffff";
    const colorField = wrapper.find(".add-box-form__color").at(0);

    colorField.simulate("change", {target: {name: "colorField", value: value}});
    expect(wrapper.state('colorField').validated).toEqual(false);
  })

  it('should call REGISTER_BOX_REQUEST and REGISTER_BOX_SUCCESS on call to register box', () => {
    fetchMock.postOnce('/boxes', mockBox);

    const store = mockStore({});

    return store.dispatch(actions.register(mockBox.name, mockBox.weight, mockBox.color, mockBox.destination, mockBox.multiplier))
    .then(() => {
      expect(store.getActions()).toEqual(
        expect.arrayContaining([
          expect.objectContaining({
            type: types.REGISTER_BOX_REQUEST,
            type: types.REGISTER_BOX_SUCCESS
          })
        ])
      )
    })
  })
});
