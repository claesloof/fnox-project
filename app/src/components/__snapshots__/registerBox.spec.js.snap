// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterBox should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RegisterBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="add-box-form"
        onSubmit={[Function]}
>
        <label
                className="block__elem add-box-form__label"
        >
                Name
        </label>
        <input
                className="block__elem add-box-form__text hover-text"
                name="nameField"
                onChange={[Function]}
                type="text"
                value=""
        />
        <label
                className="block__elem add-box-form__label"
        >
                Weight
        </label>
        <input
                className="block__elem add-box-form__text hover-text"
                name="weightField"
                onChange={[Function]}
                type="text"
                value={0}
        />
        <label
                className="block__elem add-box-form__label"
        >
                Box Color
        </label>
        <input
                className="block__elem add-box-form__color hover-pointer"
                name="colorField"
                onChange={[Function]}
                type="color"
                value="#ffff00"
        />
        <label
                className="block__elem add-box-form__label"
        >
                Destination Country
        </label>
        <select
                className="block__elem add-box-form__dropdown hover-pointer"
                name="destinationField"
                onChange={[Function]}
                value="sweden"
        >
                <option
                        className=""
                        value="sweden"
                >
                        Sweden
                </option>
                <option
                        className=""
                        value="china"
                >
                        China
                </option>
                <option
                        className=""
                        value="brazil"
                >
                        Brazil
                </option>
                <option
                        className=""
                        value="australia"
                >
                        Australia
                </option>
        </select>
        <input
                className="block__elem add-box-form__submit hover-pointer"
                disabled={true}
                type="submit"
                value="SAVE"
        />
</form>,
      "className": "block",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="block__elem add-box-form__label"
>
            Name
</label>,
          <input
            className="block__elem add-box-form__text hover-text"
            name="nameField"
            onChange={[Function]}
            type="text"
            value=""
/>,
          <label
            className="block__elem add-box-form__label"
>
            Weight
</label>,
          <input
            className="block__elem add-box-form__text hover-text"
            name="weightField"
            onChange={[Function]}
            type="text"
            value={0}
/>,
          <label
            className="block__elem add-box-form__label"
>
            Box Color
</label>,
          <input
            className="block__elem add-box-form__color hover-pointer"
            name="colorField"
            onChange={[Function]}
            type="color"
            value="#ffff00"
/>,
          <label
            className="block__elem add-box-form__label"
>
            Destination Country
</label>,
          <select
            className="block__elem add-box-form__dropdown hover-pointer"
            name="destinationField"
            onChange={[Function]}
            value="sweden"
>
            <option
                        className=""
                        value="sweden"
            >
                        Sweden
            </option>
            <option
                        className=""
                        value="china"
            >
                        China
            </option>
            <option
                        className=""
                        value="brazil"
            >
                        Brazil
            </option>
            <option
                        className=""
                        value="australia"
            >
                        Australia
            </option>
</select>,
          <input
            className="block__elem add-box-form__submit hover-pointer"
            disabled={true}
            type="submit"
            value="SAVE"
/>,
        ],
        "className": "add-box-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Name",
            "className": "block__elem add-box-form__label",
          },
          "ref": null,
          "rendered": "Name",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "block__elem add-box-form__text hover-text",
            "name": "nameField",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Weight",
            "className": "block__elem add-box-form__label",
          },
          "ref": null,
          "rendered": "Weight",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "block__elem add-box-form__text hover-text",
            "name": "weightField",
            "onChange": [Function],
            "type": "text",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Box Color",
            "className": "block__elem add-box-form__label",
          },
          "ref": null,
          "rendered": "Box Color",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "block__elem add-box-form__color hover-pointer",
            "name": "colorField",
            "onChange": [Function],
            "type": "color",
            "value": "#ffff00",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Destination Country",
            "className": "block__elem add-box-form__label",
          },
          "ref": null,
          "rendered": "Destination Country",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                className=""
                value="sweden"
>
                Sweden
</option>,
              <option
                className=""
                value="china"
>
                China
</option>,
              <option
                className=""
                value="brazil"
>
                Brazil
</option>,
              <option
                className=""
                value="australia"
>
                Australia
</option>,
            ],
            "className": "block__elem add-box-form__dropdown hover-pointer",
            "name": "destinationField",
            "onChange": [Function],
            "value": "sweden",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "sweden",
              "nodeType": "host",
              "props": Object {
                "children": "Sweden",
                "className": "",
                "value": "sweden",
              },
              "ref": null,
              "rendered": "Sweden",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "china",
              "nodeType": "host",
              "props": Object {
                "children": "China",
                "className": "",
                "value": "china",
              },
              "ref": null,
              "rendered": "China",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "brazil",
              "nodeType": "host",
              "props": Object {
                "children": "Brazil",
                "className": "",
                "value": "brazil",
              },
              "ref": null,
              "rendered": "Brazil",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "australia",
              "nodeType": "host",
              "props": Object {
                "children": "Australia",
                "className": "",
                "value": "australia",
              },
              "ref": null,
              "rendered": "Australia",
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "block__elem add-box-form__submit hover-pointer",
            "disabled": true,
            "type": "submit",
            "value": "SAVE",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="add-box-form"
          onSubmit={[Function]}
>
          <label
                    className="block__elem add-box-form__label"
          >
                    Name
          </label>
          <input
                    className="block__elem add-box-form__text hover-text"
                    name="nameField"
                    onChange={[Function]}
                    type="text"
                    value=""
          />
          <label
                    className="block__elem add-box-form__label"
          >
                    Weight
          </label>
          <input
                    className="block__elem add-box-form__text hover-text"
                    name="weightField"
                    onChange={[Function]}
                    type="text"
                    value={0}
          />
          <label
                    className="block__elem add-box-form__label"
          >
                    Box Color
          </label>
          <input
                    className="block__elem add-box-form__color hover-pointer"
                    name="colorField"
                    onChange={[Function]}
                    type="color"
                    value="#ffff00"
          />
          <label
                    className="block__elem add-box-form__label"
          >
                    Destination Country
          </label>
          <select
                    className="block__elem add-box-form__dropdown hover-pointer"
                    name="destinationField"
                    onChange={[Function]}
                    value="sweden"
          >
                    <option
                              className=""
                              value="sweden"
                    >
                              Sweden
                    </option>
                    <option
                              className=""
                              value="china"
                    >
                              China
                    </option>
                    <option
                              className=""
                              value="brazil"
                    >
                              Brazil
                    </option>
                    <option
                              className=""
                              value="australia"
                    >
                              Australia
                    </option>
          </select>
          <input
                    className="block__elem add-box-form__submit hover-pointer"
                    disabled={true}
                    type="submit"
                    value="SAVE"
          />
</form>,
        "className": "block",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="block__elem add-box-form__label"
>
              Name
</label>,
            <input
              className="block__elem add-box-form__text hover-text"
              name="nameField"
              onChange={[Function]}
              type="text"
              value=""
/>,
            <label
              className="block__elem add-box-form__label"
>
              Weight
</label>,
            <input
              className="block__elem add-box-form__text hover-text"
              name="weightField"
              onChange={[Function]}
              type="text"
              value={0}
/>,
            <label
              className="block__elem add-box-form__label"
>
              Box Color
</label>,
            <input
              className="block__elem add-box-form__color hover-pointer"
              name="colorField"
              onChange={[Function]}
              type="color"
              value="#ffff00"
/>,
            <label
              className="block__elem add-box-form__label"
>
              Destination Country
</label>,
            <select
              className="block__elem add-box-form__dropdown hover-pointer"
              name="destinationField"
              onChange={[Function]}
              value="sweden"
>
              <option
                            className=""
                            value="sweden"
              >
                            Sweden
              </option>
              <option
                            className=""
                            value="china"
              >
                            China
              </option>
              <option
                            className=""
                            value="brazil"
              >
                            Brazil
              </option>
              <option
                            className=""
                            value="australia"
              >
                            Australia
              </option>
</select>,
            <input
              className="block__elem add-box-form__submit hover-pointer"
              disabled={true}
              type="submit"
              value="SAVE"
/>,
          ],
          "className": "add-box-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Name",
              "className": "block__elem add-box-form__label",
            },
            "ref": null,
            "rendered": "Name",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "block__elem add-box-form__text hover-text",
              "name": "nameField",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Weight",
              "className": "block__elem add-box-form__label",
            },
            "ref": null,
            "rendered": "Weight",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "block__elem add-box-form__text hover-text",
              "name": "weightField",
              "onChange": [Function],
              "type": "text",
              "value": 0,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Box Color",
              "className": "block__elem add-box-form__label",
            },
            "ref": null,
            "rendered": "Box Color",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "block__elem add-box-form__color hover-pointer",
              "name": "colorField",
              "onChange": [Function],
              "type": "color",
              "value": "#ffff00",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Destination Country",
              "className": "block__elem add-box-form__label",
            },
            "ref": null,
            "rendered": "Destination Country",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  className=""
                  value="sweden"
>
                  Sweden
</option>,
                <option
                  className=""
                  value="china"
>
                  China
</option>,
                <option
                  className=""
                  value="brazil"
>
                  Brazil
</option>,
                <option
                  className=""
                  value="australia"
>
                  Australia
</option>,
              ],
              "className": "block__elem add-box-form__dropdown hover-pointer",
              "name": "destinationField",
              "onChange": [Function],
              "value": "sweden",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "sweden",
                "nodeType": "host",
                "props": Object {
                  "children": "Sweden",
                  "className": "",
                  "value": "sweden",
                },
                "ref": null,
                "rendered": "Sweden",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "china",
                "nodeType": "host",
                "props": Object {
                  "children": "China",
                  "className": "",
                  "value": "china",
                },
                "ref": null,
                "rendered": "China",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "brazil",
                "nodeType": "host",
                "props": Object {
                  "children": "Brazil",
                  "className": "",
                  "value": "brazil",
                },
                "ref": null,
                "rendered": "Brazil",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "australia",
                "nodeType": "host",
                "props": Object {
                  "children": "Australia",
                  "className": "",
                  "value": "australia",
                },
                "ref": null,
                "rendered": "Australia",
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "block__elem add-box-form__submit hover-pointer",
              "disabled": true,
              "type": "submit",
              "value": "SAVE",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
